Sources/SwiftProtobuf/BinaryDecoder.swift

```diff
@@ -46,12 +50,47 @@ internal struct BinaryDecoder: Decoder {
 
     private var complete: Bool {return available == 0}
 
-    internal init(forReadingFrom pointer: UnsafePointer<UInt8>, count: Int, extensions: ExtensionMap? = nil) {
+    internal init(
+      forReadingFrom pointer: UnsafePointer<UInt8>,
+      count: Int,
+      options: BinaryDecodingOptions,
+      extensions: ExtensionMap? = nil
+    ) {
         // Assuming baseAddress is not nil.
         p = pointer
         available = count
         fieldStartP = p
         self.extensions = extensions
+        self.options = options
+        recursionBudget = options.messageDepthLimit
+    }
+
+    internal init(
+      forReadingFrom pointer: UnsafePointer<UInt8>,
+      count: Int,
+      parent: BinaryDecoder
```

Comment:
Adding a commit to make the parent inout and then comparing that to this CL seems to be reporting everything was faster at this commit, so the inout slows things down.  So I'm not completely sure what to say here.  Do we want to get this in and then you can look?  Strangely, if I add a commit that isn't pushed off my local machine and then run comparing to the previous commit, the script always exists with a change staged and local edits.  The staged changes seem to be without my last comment and the unstagged changes seem to to be that last commit, so it's like a reverse/restore is happening via the script?